safe-room.lic

```diff
@@ -63,6 +65,33 @@ class SafeRoom
     score > @health_threshold
   end
 
+  def devour_healing(settings)
+    return unless settings.necromancer_healing['Devour']
+    return unless settings.thanatology['harvest_container']
+    
+    walk_to(settings.safe_room)
+    release_invisibility
+    devours = count_items_in_container('material', settings.thanatology['harvest_container'])
+    return unless devours > 0
+
+    stow_hands
+    return if right_hand
+    
+    while need_healing? && devours > 0
+      while mana < 40
+        echo('Waiting on mana...')
+        pause 10
+      end
+
+      result = bput("get material in my #{settings.thanatology['harvest_container']}", 'You get', 'You are already', 'What were you') unless right_hand
+      break if result =~ 'What were you'
+      DRCA.cast_spell(settings.necromancer_healing['Devour'], settings)
```

Comment:
You need to add `common-arcana` to `custom_require` at the top. If you also add `include DRCA` to the class, you don't need to prefix this function call.

Reply:
Done.